### ISSUE TITLE
Any plans to support doTransform() with non-stamped message types?

### ISSUE THREAD

**Issue by mitsudome-r** [ros2/geometry2#337]
## Feature request
In ROS1, we used to have doTransforms for message types without Headers, but they are missing in ROS2.
Are there any reasons why will omitted these functions in ROS2?

Example with geometry_msgs/Point
[ROS1](https://github.com/ros/geometry2/blob/noetic-devel/tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.h#L722) 
[ROS2](https://github.com/ros2/geometry2/blob/ce8f0070b08389c6c8c0874c4eb1c6143d893bb6/tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.h#L158)

#### Feature description
Transforms geometry messages without timestamps.

**Comment by clalancette** [comment]
The original commit that added those to https://github.com/ros/geometry2 is [here](https://github.com/ros/geometry2/commit/21e2d9378742116b9ccb7d2e95ed0963e221b231).  That was added after we forked this repository off from that one.

No fundamental objection to having those.  Contributions to add it are very welcome.

**Comment by adamkrawczyk** [comment]
Hi,

I'm doing this and create PR but wondering whether adding PoseWithCovariance without timestamp makes sense? 
 
Best

**Comment by aprotyas** [comment]
> wondering whether adding PoseWithCovariance without timestamp makes sense?

You don't necessarily need headers for all types of applications. For example, I can think of a planar path planner that would not care much about the timestamp/reference frame of goal queries expressed as `PoseWithCovariance`s.

**Comment by adamkrawczyk** [comment]
Made changes at:
https://github.com/ros2/geometry2/pull/452

Please take a look and share your thoughts

**Comment by galou** [comment]
Isn't it actually misleading to have `doTransform<geometry_msgs::msg::PoseStamped>` because two `PoseStamped` that have the same coordinates but different frames, i.e. that are supposed to express two different poses, are transformed into the same `PoseStamped`. Generally, the poses' frame is ignored completely. Removing `doTransform<geometry_msgs::msg::PoseStamped>` would break the API though.

**Comment by aprotyas** [comment]
> Isn't it actually misleading to have `doTransform<geometry_msgs::msg::PoseStamped>` because two `PoseStamped` that have the same coordinates but different frames, i.e. that are supposed to express two different poses, are transformed into the same `PoseStamped`. Generally, the poses' frame is ignored completely. Removing `doTransform<geometry_msgs::msg::PoseStamped>` would break the API though.

@galou wouldn't that be valid behavior though? As in, if you're performing a transformation, you're expressing the geometric data (pose) in a different frame, so the frame ID of the stamped pose should be updated accordingly?

**Comment by galou** [comment]
@aprotyas I agree that the resulting `frame_id` should be the same for both `PoseStamped` but the coordinates should vary. The current implementation of `doTransform<geometry_msgs::msg::PoseStamped` completely ignores `t_in.header.frame_id`.

**Comment by aprotyas** [comment]
@galou I might be misunderstanding the math here, but I'm thinking that if you do a transformation on a stamped pose, then the output of that transformation should be in the same frame as the transformation (and not the input pose), which is [what's happening](https://github.com/ros2/geometry2/blob/83fbeb70b61d83a410e184266f150b9a0215c9a9/tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.hpp#L641) in the current implementation of `doTransform<geometry_msgs::msg::PoseStamped>`.

On a tangent, I will say the choice of `t_in` might not be the best input parameter name given `t` could also stand for transform...

**Comment by CursedRock17** [comment]
Just to double check, I think this issue was resolved by the matching PR, but does it still need help in any sort of way?
